src/db/migrations/49-extract-ipfs-files-urls-from-ipfs-files.ts

```diff
@@ -0,0 +1,60 @@
+import { Knex } from 'knex';
+import _ from 'lodash';
+
+import { Table } from '../db';
+import { tableNameToViewName, updateViews } from '../migration-helpers';
+
+// 1. moves `url` <-> `cid` relationship to a new join table to support multiple urls per cid
+// 2. changes ipfsFiles table's primary key to `cid`
+export const up = async (knex: Knex) => {
+  await knex.schema.dropViewIfExists(tableNameToViewName(Table.ipfsFiles));
+  await knex.schema.alterTable(Table.ipfsFiles, table => {
+    table.dropPrimary();
+  })
+
+  const files = await knex.raw(`select * from "${Table.ipfsFiles}" where "cid" is not null`);
+  const filesByCid = _.groupBy(files.rows, 'cid')
+  const urlCids: any[] = [];
+
+  const ipfsFileUpdates = Object.keys(filesByCid).map((key) => {
+    const row = filesByCid[key];
+
+    // prepare data for ipfsFilesUrls join table
+    row.forEach((r: any) => {
+      urlCids.push({ cid: r.cid, url: r.url })
+    });
+
+    // remove url from ipfsFiles table
+    const { url, ...rest } = row[0];
+    return rest;
+  });
+
+  // repopulate ipfsFiles table
+  await knex(Table.ipfsFiles).truncate();
```

Comment:
so this deletes all records in the table, and then re-inserts them without the url?

this seems fine, though above u only get records where "cid" is not null. so the other records will be lost? i guess that should be fine?

Reply:
Yep, those are the errored ones which then get repopulated on a reindex. I've verified with a local reindex 👍